tests/RouterTest.php

```
@@ -162,6 +162,38 @@ public function testPathForWithQueryParameters()
         );
     }
 
+    public function testPathForWithQueryParametersForPath()
+    {
+        $methods = ['GET'];
+        $pattern = '/hello/{name}';
+        $callable = function ($request, $response, $args) {
+            echo sprintf('Hello %s', $args['name']);
+        };
+        $route = $this->router->map($methods, $pattern, $callable);
+        $route->setName('foo');
+
+        $this->assertEquals(
+            '/hello/josh?a=b',
+            $this->router->pathFor('foo', [], ['name' => 'josh', 'a' => 'b'])
+        );
+    }
+
+    public function testPathForWithCombinedParameters()
+    {
+        $methods = ['GET'];
+        $pattern = '/hello/{name}';
+        $callable = function ($request, $response, $args) {
+            echo sprintf('Hello %s', $args['name']);
+        };
+        $route = $this->router->map($methods, $pattern, $callable);
+        $route->setName('foo');
+
+        $this->assertEquals(
+            '/hello/josh?a=b',
```

The use case `/route/{name}` where `/hello/josh?name=test` should throw a router exception.